generators/GenerateMavenVersions.java

```diff
@@ -0,0 +1,209 @@
+import org.apache.maven.artifact.versioning.ComparableVersion;
```

Comment:
One more minor comment:

Should we put this in say `/test/semantic`? 

I chose `/test` based on https://github.com/golang-standards/project-layout#test, but maybe there's other locations for this too.

Reply:
If you don't like `generators` (at least at the top level), then probably `scripts` or `tools` (though I didn't use that because I know it does have actual go-related uses) is most suitable.

I don't think `test` is appropriate because these are not tests.

Reply:
Yeah my main nit was `generators` being in the top level.

`scripts` SGTM!